#9833 Fix integration tests problems

nollymar opened the pull request:
- Avoid the creation of an InitialContextFactoryBuilder for each testing class. Now, it is created just once and reused.
- Avoid calling TestingJndiDatasource.init() and ConfigTestHelper._setupFakeTestingContext when running tests with the server up.
- ContentletAPITest was removed from AllTestsSuite. Now, this testing class will run with right-click from any IDE instead of Tomcat environment.

nollymar committed:
#9833 Fix integration tests problems
- Avoid the creation of an InitialContextFactoryBuilder for each testing class. Now, it is created just once and reused.
- Avoid calling TestingJndiDatasource.init() and ConfigTestHelper._setupFakeTestingContext when running tests with the server up.
- ContentletAPITest was removed from AllTestsSuite. Now, this testing class will run with right-click from any IDE instead of Tomcat environment.

jdotcms commented on test/com/dotmarketing/portlets/ContentletBaseTest.java:
```diff
@@ -109,8 +109,12 @@
     public static void prepare () throws Exception {
-        TestingJndiDatasource.init();
-        ConfigTestHelper._setupFakeTestingContext();
+
```
The class should have a more generic name, such as BaseIntegrationTest.
In addition the idea is to check the state of the Integration Framework, if it is not initialized, then init it. Otherwise we will reuse the current state.

It should work if you run a method, class or suite.

dsilvam replied:
will be done here: https://github.com/dotCMS/core/issues/9853

nollymar committed:
#9833 Removing PowerMock dependency from ContentletAPITest.java
- getServletContext() was added to MockSessionRequest.java
- ContentletAPITest.java and ContentletBaseTest.java were refactored to mock MessageResources and ModuleConfig objects needed for widgetInvalidateAllLang() test

joseorsini commented:
Disregard last message. Typo on Github issue Id
